ipykernel/eventloops.py

```diff
@@ -267,9 +262,8 @@ def __init__(self, app, func):
                 self.func = func
 
             def on_timer(self):
-                loop = asyncio.get_event_loop()
                 try:
-                    loop.run_until_complete(self.func())
+                    run_sync(self.func)()
```

Comment:
This is a Windows specific code path for which we don't have tests in Spyder-kernels.

@dalthviz, could you check that this PR doesn't break the Tk backend on Windows, as described in https://github.com/spyder-ide/spyder/issues/17024?

Also @impact27, if you're available, could you take a look at these changes?

Reply:
As far as I tested seems like this PR works with the latest Spyder 5.x branch on Windows with the Tkinter graphics backend and a custom interpreter :+1:

Reply:
Thanks @dalthviz!

Reply:
Ok, let's wait until Tuesday to see if @impact27 has time to review this. If not, please go ahead and merge it @blink1073.

Reply:
Sounds good!

Reply:
Looks good to me 👍

Reply:
From what I understand run_sync uses asyncio so it should work as intended

Reply:
That's correct, thanks @impact27!